docs/index.md

```diff
@@ -57,7 +68,17 @@ class YourCoreLib(CoreLib):
         self.user = UserDataAccess(db_connection)
         ...
 ```
-
+#### `your_core_lib.yaml` Explained:
+<b>Python script or application that uses a custom CoreLib class and a SqlAlchemyConnectionRegistry class to manage database connections.</b>
+- <b>Importing necessary modules:</b>
+CoreLib from core_lib.core_lib
+SqlAlchemyConnectionRegistry from core_lib.connection.sql_alchemy_connection_registry
+#### Defining a new class YourCoreLib that inherits from CoreLib.
+  - Defining an __init__ method for YourCoreLib that takes a config argument of type DictConfig. DictConfig is a custom configuration class or type used to pass configuration parameters.
+  - Calling the parent class CoreLib's __init__ method using CoreLib.__init__(self) to initialize the base class.
```

Comment:
inorder to use tho use the core-lib observer `CoreLibListener` and to be able to mark your core-lib as started